src/aap_eda/core/models/activation.py

```diff
@@ -93,6 +117,19 @@ class Meta:
     activation_pod_id = models.TextField(null=True)
     status_message = models.TextField(null=True, default=None)
 
+    def save(self, *args, **kwargs):
+        if kwargs is None or kwargs.get("update_fields") is None:
```

Comment:
with this logic as it is, you would override any status_message by its default every time the save method is called without update_fields. You need to compare with the previous value to know if status_message should have a new default message or not.

Reply:
@hsong-rh Can we just use the elif portion of this and not use the if portion? Because we want to set the status and the status_message as a pair all the time. if the caller has not provided the status_message we provide the default like the elif portion is doing it.

Reply:
@hsong-rh I think we might also want to prevent someone from just changing the status_message without the status variable defined. If it's currently in Error state, we can't just have an update_field with status_message set to "Everything is OK". If they have to change the status_message they should change the status and the status_message.

Reply:
There was also the discussion last week that since we many have too many processes (api-worker, activation-worker, default-worker) working with the Activation and ActivationInstance we should not allow saves without update_fields, since one process can end up clobbering a previous workers updates.

Reply:
The select_for_update/atomic transaction is used in the outside block, here the activation record is locked already, and update the fields. 

@mkanoor @Alex-Izquierdo 
1. now normal `save()` without any parameters will raise a validation error;
2. `save(update_fields=["status_message"])` will raise a validation error;
3. `save(update_fields=["status"])` will also update `status_message` with default if it's none